### PR TITLE
Widgets dev harness

### PR DESCRIPTION
**Pull request by ilmarinen**
Check the README for updated instructions on how to run the Dev Harness. Since there are two, I needed to get rid of the .flaskenv file and explicitly pass in environment variable values.

### CODE REVIEW

File: development/model-comparison/app.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+    with torch.no_grad():
+        _, _, output = h1(testing_set[0][1])
+
+    h2 = MLPClassifier(9, 2)
+
+    model_comparison = ModelComparison("sweeps-fico", number_of_epochs, h1, h2, training_set,
+                                       use_ml_flow=True, device="cuda")
+
+    # CompatibilityAnalysis(
```
Comment (Nking92) [inline development/model-comparison/app.py:91]: Did you mean to leave the commented code

File: development/model-comparison/app.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+    def get_instance_label(instance_id):
+        label = dataset[instance_id][2].item()
+        return f'{label}'
+
+    model_comparison = ModelComparison("sweeps-fico", n_epochs, h1, h2, train_loader,
+                                       get_instance_image_by_id=get_instance_image,
+                                       use_ml_flow=True, device="cuda")
+
+    # CompatibilityAnalysis(sweeps_folder, n_epochs, h1, h2, train_loader, test_loader,
```
Comment (Nking92) [inline development/model-comparison/app.py:211]: Here's a little more commented code

File: README.md [hunk 3]
```diff
@@ -48,18 +48,27 @@ To run tests, make sure that you are in the project root folder and do:
...
 
+## Compatibility Analysis Widget
+
+- Open a new terminal and within the project root folder do `FLASK_ENV=development FLASK_APP=development/compatibility-analysis/app.py flask run --host 0.0.0.0 --port 5000`. This will start the Flask server for the APIs used by the widget.
+- Open a new terminal, then from within the project root folder do `npm run start-compatibility-analysis`
+- Open your browser and point it to `http://<your-ip-address>:3000`
+
+## Model Comparison Widget
+
+- Open a new terminal and within the project root folder do `FLASK_ENV=development FLASK_APP=development/model-comparison/app.py flask run --host 0.0.0.0 --port 5000`. This will start the Flask server for the APIs used by the widget.
```
Comment (Nking92) [inline README.md:67]: The equivalent command for Windows is `set FLASK_ENV=development && set FLASK_APP=development\model-comparison\app.py && flask run --host 0.0.0.0 --port 5000` if you want to include that

**Review by Nking92 — APPROVED** [review]
Looks fine, left a few nitpicky suggestions. I was able to run it on my machine without any issues.

File: development/model-comparison/app.py (hunk 1 above)
Comment (ilmarinen) [reply development/model-comparison/app.py:91]: Thanks. I will remove this.

File: README.md (hunk 3 above)
Comment (ilmarinen) [reply README.md:67]: Will do.

**Comment by ilmarinen** [comment]
@Nking92 Thank you! I made all the changes you requested!

File: README.md [hunk 4]
```diff
@@ -48,18 +48,27 @@ To run tests, make sure that you are in the project root folder and do:
...
 
-The widget can be loaded in the web browser at `localhost:3000`. It will be loaded independently from a Jupyter notebook. The APIs will be hosted at `localhost:5000`. If you are running the server within a VM, you might need to use your machine's local IP instead of `localhost`.
+The widget can be loaded in the web browser at `localhost:3000` or `<your-ip>:3000`. It will be loaded independently from a Jupyter notebook. The APIs will be hosted at `localhost:5000` or `<your-ip>:5000`.
 
 Changes to the CSS or TypeScript code will be hot loaded automatically in the browser. Flask will run in debug mode and automatically restart whenever the Python code is changed.
 
+## Compatibility Analysis Widget
+
+- Open a new terminal and within the project root folder do `FLASK_ENV=development FLASK_APP=development/compatibility-analysis/app.py flask run --host 0.0.0.0 --port 5000` on Linux or `set FLASK_ENV=development && set FLASK_APP=development\comparibility-analysis\app.py && flask run --host 0.0.0.0 --port 5000` on Windows. This will start the Flask server for the APIs used by the widget.
```
Comment (jlema) [inline README.md:61]: Little typo on the Windows command **development\comparibility-analysis\app.py** should be **development\compatibility-analysis\app.py**

**Review by jlema — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (ilmarinen) [reply README.md]: Done.

**Review by jlema — APPROVED** [review]

**Review by nushib — APPROVED** [review]

**Pull request merged**